src/main/java/com/exactpro/th2/readlog/LogPublisher.java

```diff
@@ -116,17 +126,17 @@ private void publish() throws IOException {
 
 			metaData.setId(messageId);
 
-			msgBuilder.setMetadata(metaData);						
+			msgBuilder.setMetadata(metaData);		
+			
+			anyMsgBuilder.setRawMessage(msgBuilder.build());
 		}
 
 		listOfLines.clear();
 
-        RawMessageBatch batch = builder.build();
-
-        if (batch.getMessagesCount() > 0) {
-            batchMessageRouter.sendAll(batch, QueueAttribute.PUBLISH.toString(), QueueAttribute.RAW.toString());
+        if (groupBatch.getGroupsCount() > 0) {
+            batchMessageRouter.sendAll(groupBatch.build(), QueueAttribute.PUBLISH.toString(), QueueAttribute.RAW.toString());
```

Comment:
Attributes can be removed if you use MessageRouter<RawMessageBatch> because the router will add those attributes by default